core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -824,14 +823,14 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     expiredOffsets
   }
 
-  def allOffsets: Map[TopicPartition, OffsetAndMetadata] = offsets.map { case (topicPartition, commitRecordMetadataAndOffset) =>
-    (topicPartition, commitRecordMetadataAndOffset.offsetAndMetadata)
+  def allOffsets: Map[TopicIdPartition, OffsetAndMetadata] = offsets.map { case (topicPartition, commitRecordMetadataAndOffset) =>
+    (new TopicIdPartition(Uuid.ZERO_UUID, topicPartition), commitRecordMetadataAndOffset.offsetAndMetadata)
```

Comment:
Where will the topic ID come from here?

Reply:
Until topic ids become persisted in the group metadata, we won't have the topic ID unless it is resolved using the metadata cache. I shouldn't have used the zero id here, thanks for pointing it out!

If we want to maintain the mapping performed in `getOffsets` which is using `TopicIdPartition` as keys, we could instead perform a resolution of topic id from the topic name as done in the `KafkaApis` request-handler for topic-id-aware requests before they are processed. But then we would need to consider the cases where an id cannot be found. I will think about it a bit more.

Reply:
What cases are we considering? A topic should have an ID as long as controller is ibp 2.8+

Reply:
Hi Justine, thanks for the reply. Apologies I was OOO. I was referring to exceptional errors, although not functionally expected.

Reply:
Hi Justine (@jolshan), I pushed a commit with the change to resolve the topic id in the group metadata manager when the offsets of all topic-partitions known by the coordinator are requested. I don't really like the code I wrote since it introduces one level of indirection on the valid path and one more failure path when a topic id cannot be resolved. And the implementation proposed is not very elegant. But it follows up on the discussion above. 

When the new schemas for metadata and offset records are used, the topic id will be taken from the record tag. However for all existing records, there will still be a need for reconciliation. Perhaps this reconciliation should be done when the offsets are loaded by the coordinator once the metadata data structure become topic-id-aware.

I will add a few more unit tests.